### PR TITLE
Show name and expression labels

### LINKED ISSUE #1039: Show name and expression labels on sequence flows
## Description
Show name and expression labels on sequence flows

### PR DESCRIPTION
**Pull request by tuantruong625**
The purpose of this PR is to show the name and expression labels on sequences flows.

Fixes #1039

### CODE REVIEW

File: src/components/nodes/sequenceFlow/sequenceFlow.vue [hunk 1]
```diff
@@ -79,15 +94,17 @@ export default {
...
     'node.definition': {
       handler({ startEvent: startEventId, conditionExpression }) {
-        if (!this.sourceIsGateway && !this.targetIsCallActivity) {
+        if (this.targetIsCallActivity) {
           return;
         }
 
         const startEvent = store.getters.globalProcessEvents.find(event => event.id == startEventId);
-        const newLabel = get(conditionExpression, 'body') || get(startEvent, 'name');
+        const newExpressionLabel = get(conditionExpression, 'body') || get(startEvent, 'name');
```
Comment (lisa-fehr) [inline src/components/nodes/sequenceFlow/sequenceFlow.vue]: Wonder if we can drop the `get(startEvent, 'name')`. In what scenario would this appear?

**Comment by chipit24** [comment]
The labels don't appear to be in the correct positions for me. The name is not quite in the centre, and the expression is right on the line, not under it.

Screen Shot 2020-01-23 at 2 27 37 PM

Should both labels be centred between the shapes, and be above and below the flow?

File: src/components/nodes/sequenceFlow/sequenceFlowConfig.js [hunk 2]
```diff
@@ -1 +1,2 @@
-export const expressionPosition = { offset: -25 };
+export const namePosition = { offset: { y: -20 } };
+export const expressionPosition =  { offset: { y: 40 } };
```
Comment (johnblackspear) [inline src/components/nodes/sequenceFlow/sequenceFlowConfig.js]: I suppose some fiddling with these values could be the answer to Rob's comment?

Comment (tuantruong625) [reply src/components/nodes/sequenceFlow/sequenceFlowConfig.js]: When testing with Rob, you are able to change the position of the label which caused the label to be off-centered.

File: src/components/nodes/sequenceFlow/sequenceFlow.vue (hunk 1 above)
Comment (tuantruong625) [reply src/components/nodes/sequenceFlow/sequenceFlow.vue]: Connecting a start-event to a sub-process will trigger this scenario.

**Comment by tuantruong625** [comment]
**Note**
When connecting a start event to a subprocess the association between are set in the XML but not visually represented on the sequence flow.

**Comment by chipit24** [comment]
When a gateway is connected to a sub-process, the expression doesn't show, only the name.

File: src/components/nodes/sequenceFlow/sequenceFlow.vue [hunk 3]
```diff
@@ -95,9 +98,12 @@ export default {
         const newExpressionLabel = get(conditionExpression, 'body');
         const newNameLabel = this.shapeName || get(startEvent, 'name');
 
-        if (this.targetIsCallActivity) {
+        if (this.targetIsCallActivity ) {
```
Comment (chipit24) [inline src/components/nodes/sequenceFlow/sequenceFlow.vue]:
```suggestion
        if (this.targetIsCallActivity) {
```

**Comment by velkymx** [comment]
@tuantruong625 let's remove the expression and just use the label.

**Pull request merged**